avm/src/vm/opcode.rs

```diff
@@ -2217,20 +2217,25 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     hand_mem.init_fractal(args[2]);
     None
   });
-  io!("map", |args, mut hand_mem| {
+  io!("map", |args, hand_mem| {
     Box::pin(async move {
-      let fractal = hand_mem.read_fractal(args[0]);
+      let hand_mem_ref = Arc::new(hand_mem);
+      let fractal = hand_mem_ref.read_fractal(args[0]);
       let subhandler = HandlerFragment::new(args[1], 0);
       let mut mappers = Vec::new();
       for i in 0..fractal.len() {
-        let mut hm = hand_mem.fork();
+        let mut hm = HandlerMemory::fork(Arc::clone(&hand_mem_ref));
```

Comment:
Unfortunate... :(

Reply:
I know :(
I really tried to capture this logic within `memory.rs` but it was not possible